`test_monitor.py::test_unexpected_close` is flaky

```
FAILED tests/unit_tests/ensemble_evaluator/test_monitor.py::test_unexpected_close - Failed: DID NOT RAISE <class 'websockets.exceptions.ConnectionClosedOK'>
```
https://github.com/equinor/ert/actions/runs/9465031215/job/26073699548?pr=8123

--- a/tests/unit_tests/ensemble_evaluator/test_monitor.py
+++ b/tests/unit_tests/ensemble_evaluator/test_monitor.py
@@ -64,9 +64,11 @@ async def test_unexpected_close(unused_tcp_port):
     )
 
     set_when_done = asyncio.Event()
+    socket_closed = asyncio.Event()
 
     async def mock_ws_event_handler(websocket):
         await websocket.close()
+        socket_closed.set()
 
     websocket_server_task = asyncio.create_task(
         _mock_ws(set_when_done, mock_ws_event_handler, ee_con_info)
@@ -76,6 +78,7 @@ async def test_unexpected_close(unused_tcp_port):
         # but no attempt on resubmitting
         # since connection closed via websocket.close
         with pytest.raises(ConnectionClosedOK):
+            await socket_closed.wait()
             await monitor.signal_done()
 
     set_when_done.set()

